### PR TITLE
COMP: Fixed Cpp Warnings

### PR DESCRIPTION
**Pull request by andrei-sandor**
Fixed the following warnings

-Wimplicit-fallthrough
-Wzero-as-null-pointer-constant
-Wdeprecated-dynamic-exception-spec

### CODE REVIEW

File: Modules/Core/Common/include/itkThreadSupport.h [hunk 1]
```diff
@@ -51,7 +51,7 @@ using FastMutexType = pthread_mutex_t;
 using ThreadFunctionType = void * (*)(void *);
 using ThreadProcessIdType = pthread_t;
 #  if !defined(ITK_FUTURE_LEGACY_REMOVE)
-constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = 0;
+constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = nullptr;
```
Comment (dzenanz) [inline Modules/Core/Common/include/itkThreadSupport.h:54]: I am uncomfortable with this change. Where does the warning appear?

Comment (andrei-sandor) [reply Modules/Core/Common/include/itkThreadSupport.h:54]: The warning appears on MacOS. pthread_t is a pointer. However, on Linux, it is an int. We will remove the change.

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
@andrei-sandor thank you!

File: Modules/Core/Common/include/itkThreadSupport.h (hunk 1 above)
Comment (N-Dekker) [reply Modules/Core/Common/include/itkThreadSupport.h]: @andrei-sandor @dzenanz If the warning still bothers you, I would suggest using `{}` instead. `{}` should be universal, so it should work well for both pointers and integers:

    constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID{};

**Comment by andrei-sandor** [comment]
> @andrei-sandor thank you!

@dzenanz @thewtex. I thank you. It was a pleasure to write PRs for ITK. I learned a lot form the code and from your valuable comments.

**Pull request merged**